Subject: [PATCH] kdump: Fix for machine checkstop on DMA fault
From: Haren Myneni <haren@us.ibm.com>
Message-ID: <44222462.2030103@us.ibm.com>
Date: Wed, 22 Mar 2006 20:30:26 -0800

Some machines checkstop on dma protection fault for ongoing DMA left
in the first kernel. Since, we do not shutdown devices before the kdump
boot, let them continue DMA to old kernel space.

not final yet, see
http://ozlabs.org/pipermail/linuxppc-dev/2006-March/021537.html

Bug 149717 - LTC21418-USB controller initialization issues in kdump boot
Bug 157018 - LTC21952-kdump boot is not successful on JS21
Bug 158183 - LTC21562-e100 initialization issues during kdump boot

without it, the kdump kernel hangs very very early, before printing the Linux version
banner.
with it, kdump boot via sysrq c is successful.
but no keyboard input accepted (this needs a different patch)

Signed-off-by: Olaf Hering <olh@suse.de>
---
 arch/powerpc/kernel/iommu.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6.16-olh-kdump/arch/powerpc/kernel/iommu.c
===================================================================
--- linux-2.6.16-olh-kdump.orig/arch/powerpc/kernel/iommu.c
+++ linux-2.6.16-olh-kdump/arch/powerpc/kernel/iommu.c
@@ -427,8 +427,10 @@ struct iommu_table *iommu_init_table(str
 	tbl->it_largehint = tbl->it_halfpoint;
 	spin_lock_init(&tbl->it_lock);
 
+#ifndef CONFIG_CRASH_DUMP
 	/* Clear the hardware table in case firmware left allocations in it */
 	ppc_md.tce_free(tbl, tbl->it_offset, tbl->it_size);
+#endif
 
 	if (!welcomed) {
 		printk(KERN_INFO "IOMMU table initialized, virtual merging %s\n",
